### ISSUE TITLE
Parallelization causes test cases to stop respecting fixture's apartment state

### ISSUE THREAD

**Issue by jnm2** [nunit/nunit#2388]
Affected versions: 3.7.0–3.8.0

Fails:

```c#
[assembly: Parallelizable(ParallelScope.Children)]

[Apartment(ApartmentState.STA)]
public class StaFixture
{
    [TestCase(1)]
    [TestCase(2)]
    public void Test_cases_should_inherit_apartment_from_fixture(int n)
    {
        Assert.That(Thread.CurrentThread.GetApartmentState(), Is.EqualTo(ApartmentState.STA));
    }
}
```

Also fails if `[Parallelizable(ParallelScope.Children)]` is specified on the fixture or the method.

Does not fail if nothing is parallelizable.

Does not fail if the apartment is specified on the method:
```c#
[assembly: Parallelizable(ParallelScope.Children)]

public class StaFixture
{
    [TestCase(1)]
    [TestCase(2)]
    [Apartment(ApartmentState.STA)]
    public void Test_cases_should_inherit_apartment_from_fixture(int n)
    {
        Assert.That(Thread.CurrentThread.GetApartmentState(), Is.EqualTo(ApartmentState.STA));
    }
}
```

Does not fail for non-parameterized test methods.

**Comment by jnm2** [comment]
Affected versions: 3.7.0–3.8.0

**Comment by jnm2** [comment]
@CharliePoole, you're the most knowledgeable here. Are you in a position to take this on?

**Comment by CharliePoole** [comment]
It's pretty obvious where it is happening. Essentially, apartment state is similar to parallelism in that it is determined by both the test itself and the containing tests from which it "inherits"* the behavior.

I'm tied up for the moment but I put my name on it to get to in a week or so.

========

*I put "inheritance" in quotes because it has nothing to do with either test object or attribute inheritance. One of those overloaded terms! 😸

**Comment by CharliePoole** [comment]
I'm going away on a family trip for a couple of weeks so I'm taking my name off this to free it up for somebody else to work on.

**Comment by CharliePoole** [comment]
I'll add this to the PR I'm working on.

**Comment by CharliePoole** [comment]
Right, that's why I threw it back in the pond. It's not exactly a parallelization error, or not in the same way as the other errors are due to parallelization. It's simply that the ParallelWorkItemDispatcher, when dispatching the suite that wraps the individual cases looses the information that everything should be run in parallel. I think there's another issue that relates to this, which someone is working on. Can't find it at the moment though.
